packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -59,11 +83,53 @@ const {
     activeOption.value = firstMatch.id;
   },
 });
+
+const loadingMode = computed(() => {
+  if (typeof props.loadingMode === "object") return props.loadingMode.mode;
+  return props.loadingMode;
+});
+
+const loadMoreButtonLabel = computed(() => {
+  if (typeof props.loadingMode === "object" && props.loadingMode.mode === "button") {
```

Comment:
```suggestion
  if (loadingMode.value === "button") {
```
if typescript is smart enough to realize that it then must be an object ^^